modules/app/recording/recording.tsx

```diff
@@ -102,17 +125,49 @@ const Recording = () => {
     }
   };
 
-  const handleSubmit = () => {
-    // Handle submission logic here
-    console.log("Recording submitted");
+  const handleSubmit = async () => {
+    setIsLoading(true);
+
+    if (audioRef.current && audioUrl) {
```

Comment:
if-else, and if-nested-if is already spagetthi code. 
 
 Negative if first,  check the error conditions if (error) {  warning  , return } and keep the code flat